File: app/src/main/java/com/example/fantastiqa/GameState/Road.java
package com.example.fantastiqa.GameState;

public class Road extends Area{
    public CreatureCard creature;
    public boolean gem;

    public Road(){

    }

    @Override
    public String getName() {
        return creature.name;
    }

    public Road(CreatureCard creature, boolean gem) {
        this.creature = creature;
        this.gem = gem;
    }

    public Region getConnectedRegion(Region starter){
        Region reg0 = (Region)adjacenies.get(0);
        return reg0 == starter ?
                (Region)adjacenies.get(1) :
                reg0;
    }

}

File: app/src/main/java/com/example/fantastiqa/gameState/Road.kt
package com.example.fantastiqa.gameState

import java.util.UUID

/**
 * Immutable Road representation.
 */
data class Road(
    @JvmField val creature: CreatureCard? = null,
    @JvmField val gem: Boolean = false,
    @JvmField val id: String = UUID.randomUUID().toString()
) : Area() {
    
    fun getName(): String? = creature?._name

    fun getConnectedRegion(starter: Region?, board: Board): Region? {
        val adjacent = board.getAdjacentAreas(starter!!)
        // Logic depends on finding the other side of this road in the board
        return adjacent.find { it?.first == this }?.second as? Region
    }
}
